lib/req/steps.ex

```diff
@@ -596,6 +596,91 @@ defmodule Req.Steps do
     plug.(conn)
   end
 
+  defmacrop aws_signature_loaded? do
+    if Code.ensure_loaded?(:aws_signature) do
+      true
+    else
+      quote do
+        Code.ensure_loaded?(:aws_signature)
+      end
+    end
+  end
+
+  @default_aws_region_for_global_services "us-east-1"
+
+  @doc """
+  Sign the request using AWS Signature V4.
+
+  ## Request Options
+
+    * `:aws_sigv4` - if set, the AWS credentials to sign request.
+
+      * `:access_key_id` - the aws access key id from AWS.
+
+      * `:secret_access_key` - the secret access key from AWS.
+
+      * `:service` - the AWS service.
+
+      * `:region` - if set, sets the region for AWS service. If not set,
+        will use the default region for global service (#{@default_aws_region_for_global_services}).
+        See more here: https://docs.aws.amazon.com/general/latest/gr/sigv4_changes.html.
+
+      * `:options` - if set, sets the custom options to `:aws_signature.sign_v4/10`.
+
+  ## Examples
+
+      iex> aws_opts = [
+      ...>   access_key_id: "my access key id",
+      ...>   secret_access_key: "my secret access key",
+      ...>   service: "s3",
+      ...>   region: "us-east-1"
+      ...> ]
+      iex> req = Req.new(url: URI.parse("https://ossci-datasets.s3.amazonaws.com"), aws_sigv4: aws_opts)
+      iex> Req.get!(req).body
+      [
+        "mnist/",
```

Comment:
This would only work if user has req_s3 installed. Sorry i wasnt clear, i think the primary use case is making uploads as these need to be signed. If we could have a Req one liner that can do an s3 upload I think that would be useful, wdyt?

Reply:
Ok, I used my own credentials (before pushing) and the put object worked pretty well

![Screenshot_20220601_124611](https://user-images.githubusercontent.com/6402997/171445686-8841d37f-c888-4653-be85-f51b265d6654.png)